x-pack/plugins/ml/public/application/datavisualizer/file_based/components/combined_fields/utils.ts

```diff
@@ -0,0 +1,166 @@
+  mappings: Mappings,
+  combinedFields: CombinedField[]
+): Mappings {
+  const updatedMappings = { ...mappings };
+  combinedFields.forEach((combinedField) => {
+    updatedMappings.properties[combinedField.combinedFieldName] = {
+      type: combinedField.mappingType,
+    };
+  });
+  return updatedMappings;
+}
+
+export function removeCombinedFieldsFromMappings(
+  mappings: Mappings,
+  combinedFields: CombinedField[]
+) {
+  const updatedMappings = { ...mappings };
+  combinedFields.forEach((combinedField) => {
+    delete updatedMappings.properties[combinedField.combinedFieldName];
+  });
+  return updatedMappings;
+}
+
+export function addCombinedFieldsToPipeline(
+  pipeline: IngestPipeline,
+  combinedFields: CombinedField[]
+) {
+  const updatedPipeline = _.cloneDeep(pipeline);
+  combinedFields.forEach((combinedField) => {
+    updatedPipeline.processors.push({
+      set: {
+        field: combinedField.combinedFieldName,
+        value: combinedField.fieldNames
+          .map((fieldName) => {
+            return `{{${fieldName}}}`;
+          })
+          .join(combinedField.delimiter),
+      },
+    });
+  });
+  return updatedPipeline;
+}
+
+export function removeCombinedFieldsFromPipeline(
+  pipeline: IngestPipeline,
+  combinedFields: CombinedField[]
+) {
+  return {
+    ...pipeline,
+    processors: pipeline.processors.filter((processor) => {
+      return 'set' in processor
+        ? !combinedFields.some((combinedField) => {
+            return processor.set.field === combinedField.combinedFieldName;
+          })
+        : true;
+    }),
+  };
+}
+
+export function isWithinLatRange(
+  fieldName: string,
+  fieldStats: FindFileStructureResponse['field_stats']
+) {
+  return (
+    fieldName in fieldStats &&
+    'max_value' in fieldStats[fieldName] &&
+    fieldStats[fieldName]!.max_value! <= 90 &&
+    'min_value' in fieldStats[fieldName] &&
+    fieldStats[fieldName]!.min_value! >= -90
+  );
+}
+
+export function isWithinLonRange(
+  fieldName: string,
+  fieldStats: FindFileStructureResponse['field_stats']
+) {
+  return (
+    fieldName in fieldStats &&
+    'max_value' in fieldStats[fieldName] &&
+    fieldStats[fieldName]!.max_value! <= 180 &&
+    'min_value' in fieldStats[fieldName] &&
+    fieldStats[fieldName]!.min_value! >= -180
+  );
+}
+
+export function createGeoPointCombinedField(
+  latField: string,
+  lonField: string,
+  geoPointField: string
+): CombinedField {
+  return {
+    mappingType: 'geo_point',
+    delimiter: ',',
+    combinedFieldName: geoPointField,
+    fieldNames: [latField, lonField],
+  };
+}
+
+export function getNameCollisionMsg(name: string) {
+  return i18n.translate('xpack.ml.fileDatavisualizer.nameCollisionMsg', {
+    defaultMessage: '"{name}" already exists, please provide a unique name',
+    values: { name },
+  });
+}
+
+function getGeoPointField(results: FindFileStructureResponse) {
+  const latField = results.column_names.find((columnName) => {
```

Comment:
The Import page is hanging for me with a lot of log files - I guess when it doesn't find any candidates for combined fields?

![image](https://user-images.githubusercontent.com/7405507/93458800-b4503080-f8d8-11ea-8d7b-0c93f9ac8339.png)